include/solvers/heat_transfer.h

```diff
@@ -186,110 +223,162 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
+   * @brief Set-up the initial conditions associated with the physics.
+   * Generally, physics only support imposing nodal values, but some physics
+   * additionnaly support the use of L2 projection or steady-state solutions.
    */
   void
   set_initial_conditions() override;
 
   /**
-   * @brief Update non zero constraints if the boundary is time-dependent
+   * @brief Update non zero constraints if the boundary is time-dependent.
    */
   void
   update_boundary_conditions() override;
 
   /**
-   * @brief Call for the solution of the linear system of equation using a strategy appropriate
-   * to the auxiliary physics
+   * @brief Call for the solution of the linear system of equation using a
+   * strategy appropriate to the auxiliary physics.
    *
-   * @param initial_step Provides the linear solver with indication if this solution is the first
-   * one for the system of equation or not
+   * @param initial_step Provide the linear solver with indication if this
+   * solution is the first one for the system of equation or not.
    *
-   * @param renewed_matrix Indicates to the linear solve if the system matrix has been recalculated or not
+   * @param renewed_matrix Indicate to the linear solve if the system matrix
+   * has been recalculated or not.
    */
   void
   solve_linear_system(const bool initial_step,
                       const bool renewed_matrix = true) override;
 
   /**
-   * @brief Getter methods to get the private attributes for the physic currently solved
-   * NB : dof_handler and present_solution are passed to the multiphysics
-   * interface at the end of the setup_dofs method
+   * @brief Getter method to access the private attribute dof_handler for the
+   * physic currently solved. NB : dof_handler is now passed to the
+   * multiphysics interface at the end of the setup_dofs method.
+   *
+   * @return A list of the indices of the degrees of freedom.
    */
   const DoFHandler<dim> &
   get_dof_handler() override
   {
     return dof_handler;
   }
+
+  /**
+   * @brief Getter method to access the private attribute evaluation_point for
+   * the physic currently solved.
+   *
+   * @return The point at which the evaluation is performed.
+   */
   GlobalVectorType &
   get_evaluation_point() override
   {
     return evaluation_point;
   }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * local_evaluation_point for the physic currently solved.
+   *
+   * @return The local evaluation point. Ghosts cells are not considered in
+   * this evaluation.
+   */
   GlobalVectorType &
   get_local_evaluation_point() override
   {
     return local_evaluation_point;
   }
+
+  /**
+   * @brief Getter method to access the private attribute
+   * newton_update for the physic currently solved.
+   *
+   * @return The direction \f$ \delta u^n \f$ used to perform the newton iteration
+   * in \f$ u^{n+1} = u^n + \alpha ^n \delta u ^n \f$. Where \f$ \alpha ^n \f$
```

Comment:
That's maybe a lot of information. The Newton solver should contain the details regarding the Newton update itself :)